smartsim/wlm/slurm.py

```diff
@@ -253,6 +253,8 @@ def _get_alloc_cmd(nodes: int,
                     salloc_args += [prefix + opt, str(val)]
                 else:
                     salloc_args += ["=".join((prefix + opt, str(val)))]
+        else:
+            raise ValueError("Expecting time, nodes, account as an argument")
```

Comment:
Should we tighten the looping by shifting validation outside of it and only leaving "list building logic" inside?

e.g. 

```
arguments = {key for key in options.keys()}
valid = {"t", "time", "N", "nodes", "A", "account"}

if invalids := arguments - valid:
   raise ValueError("Expecting time, nodes, account as an argument. Also received: {invalids}")

```